src/mon/MonmapMonitor.cc

```diff
@@ -35,6 +38,36 @@ static ostream& _prefix(std::ostream *_dout, Monitor *mon) {
 		<< ").monmap v" << mon->monmap->epoch << " ";
 }
 
+namespace ceph::mon_cmds {
+
+mon_cmd_ctx::mon_cmd_ctx(MonOpRequestRef op,
+                         MonmapMonitor& monmap_monitor,
+                         Monitor& mon)
+ : op(op),
+   monmap_monitor(monmap_monitor),
+   mon(mon),
+   mmon_command(*(static_cast<MMonCommand *>(op->get_req())))
+{
+  auto& m = mmon_command;
+
+  if (!cmdmap_from_json(m.cmd, &cmdmap, ss))
+   throw ceph::mon_cmds::failure(-EINVAL, ss.str());
```

Comment:
I know I mentioned std::error_code before. In this case we actually do need an integer to stick in the reply for clients.

(Though at some point if we ever redo the message types I'd like to switch to a subsystem, error code pair so we don't have to pun on POSIX errors all the time for everything.)

Reply:
+1 to avoiding "punning" POSIX errors-- I don't really feel like that's what they're for, especially because in C++ the extra values are "free". I can understand what might have been the motivation behind taking such an approach, but it's for sure not in line with my own sensibilities.